### PR TITLE
Added example ascan rule for user defined attacks

### PR DESCRIPTION
**Pull request by psiinon**

### CODE REVIEW

File: active/User defined attacks.js [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+	"The script whose uid is",
+	"Type mismatch",
+	"Unable to jump to row",
+	"Unclosed quotation mark before the character string",
+	"unexpected end of SQL command",
+	"unexpected error",
+	"Unterminated string constant",
+	"Warning: mysql_query",
+	"Warning: pg_connect",
+	//"Warning: pg_connect\(\): Unable to connect to PostgreSQL server: FATAL",
```
Comment (kingthorin) [inline active/User defined attacks.js]: Drop this comment?

File: active/User defined attacks.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+		// Dont exceed recommended number of attacks for strength
+		// feel free to disable this locally ;)
+		if (i > max_attacks) {
+			return
+		}
+		// Copy requests before reusing them
+		msg = msg.cloneRequest();
+
+		// setParam (message, parameterName, newValue)
+		//println('Attacking with ' + attacks[i])
```
Comment (kingthorin) [inline active/User defined attacks.js]: Drop this comment?

File: active/User defined attacks.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+		// Add any generic checks here, eg
+		var code = msg.getResponseHeader().getStatusCode()
+		if (code >= 500 && code < 600) {
+			raiseAlert(as, msg, param, attacks[i], code)
+			// Only raise one alert per node
+			return
+		}
+
+		var body = msg.getResponseBody().toString()
+		var re = new RegExp(evidence.join("|"), "i");
```
Comment (kingthorin) [inline active/User defined attacks.js]: I'm sure it probably doesn't matter for JS, but should be consistent through in using trailing semi-colon or not.

File: active/User defined attacks.js [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+		as.setParam(msg, param, attacks[i]);
+		
+		// sendAndReceive(msg, followRedirect, handleAntiCSRFtoken)
+		as.sendAndReceive(msg, false, false);
+
+		// Add any generic checks here, eg
+		var code = msg.getResponseHeader().getStatusCode()
+		if (code >= 500 && code < 600) {
+			raiseAlert(as, msg, param, attacks[i], code)
+			// Only raise one alert per node
```
Comment (kingthorin) [inline active/User defined attacks.js]: Should this comment say "param"? (Following with the comment in the scanNode function above.)

File: active/User defined attacks.js [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+}
+
+function raiseAlert(as, msg, param, attack, evidence) {
+	// Replace with more suitable information
+	// raiseAlert(risk, int confidence, String name, String description, String uri, 
+	//		String param, String attack, String otherInfo, String solution, String evidence, 
+	//		int cweId, int wascId, HttpMessage msg)
+	// risk: 0: info, 1: low, 2: medium, 3: high
+	// confidence: 0: falsePositive, 1: low, 2: medium, 3: high, 4: confirmed
+	//println('Raising an alert... ' + evidence)
```
Comment (kingthorin) [inline active/User defined attacks.js]: Drop this comment? (Just the one line)

File: active/User defined attacks.js [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+	as.raiseAlert(1, 1, 'Active Vulnerability title', 'Full description',
```
Comment (kingthorin) [inline active/User defined attacks.js]: Capital on "Title", just for consistency?

File: active/User defined attacks.js [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+ * @param {string} value - the original parameter value.
+ */
+function scan(as, msg, param, value) {
+	// Debugging can be done using println like this
+	//println('scan called for url=' + msg.getRequestHeader().getURI().toString() + 
+	//	' param=' + param + ' value=' + value);
+	
+	var max_attacks = attacks.length	// No limit for the "INSANE" level ;)
+	
+	if (as.getAlertThreshold() == "LOW") {
```
Comment (thc202) [inline active/User defined attacks.js]: Should be `getAttackStrength()`?

**Comment by psiinon** [comment]
Thanks - think I've fixed all those now

**Comment by thc202** [comment]
Thanks!

**Pull request merged**